pkg/kubelet/kubelet_pods.go

```diff
@@ -986,6 +986,15 @@ func (kl *Kubelet) filterOutInactivePods(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 
+		// If a pod can be considered as terminated on k8s API server, it should be always
+		// excluded from the list of pods.
+		// This is needed to avoid insufficient resources errors when kubelet admits new pods.
+		if s, ok := kl.statusManager.GetPodStatus(p.UID); ok {
```

Comment:
this change can solve this problem, but we should consider that whether we really need the !kl.podWorkers.IsPodTerminationRequested(p.UID) check on line 985, since this behavior is not consitent with scheduler
/cc @smarterclayton

Reply:
I agree that it is a better solution to remove the check if we can remove it.